pallets/subtensor/src/block_step.rs

```diff
@@ -82,6 +84,38 @@ impl<T: Config> Pallet<T> {
         LoadedEmission::<T>::get(netuid).unwrap()
     }
 
+    // Checks the current block emission and
+    // updates the block emission if the recalculated value is lower
+    //
+    pub fn check_halving() {
+        const ORIGINAL_EMISSION: f64 = 1_000_000_000.0; //Rao => 1 Tao
+        let current_issuance: f64 = Self::get_total_issuance() as f64;
+
+        let emission_percentage: f64 = Self::get_emission_from_issuance(current_issuance);
+        let new_emission_float: f64 = ORIGINAL_EMISSION * emission_percentage;
+        let new_emission: u64 = new_emission_float as u64;
+
+        let current_emission: u64 = Self::get_block_emission();
+
+        if current_emission > new_emission {
+            Self::set_block_emission(new_emission);
+        }
+    }
+
+    // Calculates the emission rate based on the total issuance,
+    // where the emission decreases as the total issuance approaches the supply cap.
+    //
+    pub fn get_emission_from_issuance(total_issuance: f64) -> f64 {
+        const TOTAL_SUPPLY: f64 = 21_000_000_000_000_000.0; //Rao => 21_000_000 Tao
```

Comment:
Same comment as `ORIGINAL_EMISSION` -- define this as a config item, and have the runtime supply this number.